drivers/serial/uart_ns16550.c

```diff
@@ -320,6 +326,7 @@ static int uart_ns16550_configure(struct device *dev,
 
 	unsigned int old_level;     /* old interrupt lock level */
 	u8_t mdc = 0U;
+	k_spinlock_key_t key = k_spin_lock(&dev_data->lock);
```

Comment:
Since there are multiple error return locations and all are within a locked region a cleaner (IMO) idiom would be something like:
```
int rv = 0;
// stuff here
key = k_spin_lock(&dev_data->lock);
// more stuff here
if (some_error_detected) {
  rv = -ENOTSUP;
  goto out;
}
// ....
out:
   k_spin_unlock(&dev_data->lock, key);

   return rv;
```

Reply:
Thanks for catching that unlock, and I'm all for cleaner code. I'll update it.

Reply:
fwiw At first I was a little unsure if we should use goto due to secure coding practice guidelines, but in this context looks like we are ok. At least according to MISRA-C which advises against the use of goto (Rule 15.1) but sets requirements if goto is put into use: 
Rule 15.2 (Required): The goto statement shall jump to a label declared later in the same function
Rule 15.3 (Required): Any label referenced by a goto statement shall be declared in the same block, or in any block enclosing the goto statement